Framework/DataHandling/src/LoadMask.cpp

```diff
@@ -39,6 +39,234 @@ using namespace Mantid::Kernel;
+* @param pairs   -- vector of obects, defined as pairs, in the form min,max
+*                   value
+*/
+template <typename T>
+void parseRangeText(const std::string &inputstr, std::vector<T> &singles,
+                    std::vector<T> &pairs) {
+  // 1. Split ','
+  std::vector<std::string> rawstrings;
+  boost::split(rawstrings, inputstr, boost::is_any_of(","),
+               boost::token_compress_on);
+
+  for (auto &rawstring : rawstrings) {
+    // a) Find '-':
+    boost::trim(rawstring);
+    bool containDash(true);
+    if (rawstring.find_first_of("-") == std::string::npos) {
+      containDash = false;
+    }
+
+    // Process appropriately
+    if (containDash) { // 4. Treat pairs
+      std::vector<std::string> ptemp;
+      boost::split(ptemp, rawstring, boost::is_any_of("-"),
+                   boost::token_compress_on);
+      if (ptemp.size() != 2) {
+        std::string error =
+            "Range string " + rawstring + " has a wrong format!";
+        throw std::invalid_argument(error);
+      }
+      // b) parse
+      T intstart = boost::lexical_cast<T>(ptemp[0]);
+      T intend = boost::lexical_cast<T>(ptemp[1]);
+      if (intstart >= intend) {
+        std::string error =
+            "Range string " + rawstring + " has wrong order of detectors ID!";
+        throw std::invalid_argument(error);
+      }
+      pairs.push_back(intstart);
+      pairs.push_back(intend);
+
+    } else { // 3. Treat singles
+      T itemp = boost::lexical_cast<T>(rawstring);
+      singles.push_back(itemp);
+    }
+  } // ENDFOR i
+}
+
+/*
+* Parse a line in an ISIS mask file string to vector
+* Combination of 5 types of format for unit
+* (1) a (2) a-b (3) a - b (4) a- b (5) a- b
+* separated by space(s)
+* @param  ins    -- input string in ISIS ASCII format
+* @return ranges -- vector of a,b pairs converted from input
+*/
+void parseISISStringToVector(const std::string &ins,
+                             std::vector<Mantid::specnum_t> &ranges) {
+  // 1. Split by space
+  std::vector<string> splitstrings;
+  boost::split(splitstrings, ins, boost::is_any_of(" "),
+               boost::token_compress_on);
+
+  // 2. Replace a-b to a - b, remove a-b and insert a, -, b
+  bool tocontinue = true;
+  size_t index = 0;
+  while (tocontinue) {
+    // a) Determine end of loop.  Note that loop size changes
+    if (index == splitstrings.size() - 1) {
+      tocontinue = false;
+    }
+
+    // b) Need to split?
+    vector<string> temps;
+    boost::split(temps, splitstrings[index], boost::is_any_of("-"),
+                 boost::token_compress_on);
+    if (splitstrings[index].compare("-") == 0 || temps.size() == 1) {
+      // Nothing to split
+      index++;
+    } else if (temps.size() == 2) {
+      // Has a '-' inside.  Delete and Replace
+      temps.insert(temps.begin() + 1, "-");
+      splitstrings.erase(splitstrings.begin() + index);
+      for (size_t ic = 0; ic < 3; ic++) {
+        if (temps[ic].size() > 0) {
```

Comment:
not sure empty is suitable here -- you probably always have data in a string. Left as it was as its not my code.